lib/process_settings/file_monitor.rb

```diff
@@ -24,6 +25,26 @@ def initialize(file_path, logger:, environment: nil)
       start_internal(enable_listen_thread?(environment))
     end
 
+    def start_watchdog_thread(file_path)
+      @watchdog_thread and raise ArgumentError, "watchdog thread already running!"
+      @watchdog_thread = Thread.new do
+        watchdog = ProcessSettings::Watchdog.new(file_path)
+        loop do
+          ExceptionHandling.ensure_safe("ProcessSettings::Watchdog thread") do
+            sleep(1.minute)
+            watchdog.check
+          end
+        end
+      end
+    end
+
+    def stop_watchdog_thread
+      if @watchdog_thread
+        Thread.kill(@watchdog_thread)
+        @watchdog_thread = nil
```

Comment:
BTW I think the instance method of `kill` is usually preferable. In which case this code can read:
```
@watchdog_thread&.kill
@watchdog_thread = nil
```